src/threads.c

```diff
@@ -38,7 +38,9 @@ static struct task_struct *next_tid(struct task_struct *start) {
 
 void recover_threads_snapshot(struct task_data *data) {
 
-  struct task_struct *t = get_task_struct(data->tsk->group_leader);
+  struct task_struct *t = data->tsk->group_leader;
+  atomic_inc(&(data->tsk->group_leader)->usage);
+
```

Comment:
this change makes 5.x kernels fail.
you have to add an #if for the kernel version and find the right cut-off point :)

Reply:
`#include <linux/version.h>
`
...
```
void recover_threads_snapshot(struct task_data *data) {

#if LINUX_VERSION_CODE < KERNEL_VERSION(5,1,0)
  struct task_struct *t = data->tsk->group_leader;
  atomic_inc(&t->usage);
#elif LINUX_VERSION_CODE < KERNEL_VERSION(5,4,0)
  struct task_struct *t = data->tsk->group_leader;
  refcount_inc(&t->usage);
#else
  struct task_struct *t = get_task_struct(data->tsk->group_leader);
#endif

```
should be like this.

Reply:
even better, works universally (without `#ifdef`s) with these lines instead:
```
  struct task_struct *t = data->tsk->group_leader;
  get_task_struct(t);
```
Checked with Ubuntu 18.04 (kernel 4.15.0), Arch Linux (kernel 5.8.7)